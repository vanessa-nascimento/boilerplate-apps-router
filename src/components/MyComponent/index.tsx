import * as S from './styles'

const MyComponent = () => (
  <S.Wrapper>
    <h1>MyComponent</h1>
  </S.Wrapper>
)

export default MyComponent
