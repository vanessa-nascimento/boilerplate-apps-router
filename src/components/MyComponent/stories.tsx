import { Meta, StoryObj } from '@storybook/react'
import MyComponent from '.'

export default {
  title: 'MyComponent',
  component: MyComponent
} as Meta

export const Default: StoryObj = {}
