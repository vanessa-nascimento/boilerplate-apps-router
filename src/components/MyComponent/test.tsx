import { render, screen } from '@testing-library/react'

import MyComponent from '.'

describe('<MyComponent />', () => {
  it('should render the heading', () => {
    const { container } = render(<MyComponent />)

    expect(
      screen.getByRole('heading', { name: /MyComponent/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
