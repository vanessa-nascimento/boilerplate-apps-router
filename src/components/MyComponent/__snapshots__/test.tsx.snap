// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyComponent /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    MyComponent
  </h1>
</main>
`;
